sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/JavaTypeInference.scala

```diff
@@ -131,9 +130,16 @@ object JavaTypeInference {
     }
   }
 
+  def getJavaBeanPropertiesWithGetters(beanClass: Class[_]): Array[PropertyDescriptor] = {
+    val beanInfo = Introspector.getBeanInfo(beanClass)
+    beanInfo.getPropertyDescriptors
+      .filterNot(_.getName == "class").filter(p => p.getReadMethod != null)
+  }
+
   private def getJavaBeanProperties(beanClass: Class[_]): Array[PropertyDescriptor] = {
```

Comment:
Can this call the new method above and additionally filter for a setter?
The naming ends up a bit funny because the first method sounds like it returns a subset of the second method, but it's the other way around. The first one returns "readable properties"? and the second returns "read-write properties"? not sure, might be worth clarifying